calendar/modules/listener/watch_scheduler.bal

```diff
@@ -14,20 +14,46 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/lang.runtime;
 import ballerina/task;
 import ballerina/time;
 
 class Job {
     *task:Job;
     private Listener 'listener;
+    private int retryCount = 0;
+    private int retryScheduleCount = 0;
 
     isolated function init(Listener 'listener) {
         self.'listener = 'listener;
     }
 
     public isolated function execute() {
-        checkpanic self.'listener.registerWatchChannel();
-        checkpanic self.scheduleNextChannelRenewal();
+        error? err = self.'listener.registerWatchChannel();
+        if (err is error) {
+            if (self.retryCount < 10) {
+                runtime:sleep(5);
+                self.retryCount += 1;
```

Comment:
Shall we print a log saying we are retrying to register watch channel - retry count is = x

Reply:
Fixed in bb0c08e9a7f7da0b076f584b770b90e7d7a06aa4